container/state.go

```diff
@@ -385,6 +385,17 @@ func (s *State) IsDead() bool {
 // closes the internal waitRemove channel to unblock callers waiting for a
 // container to be removed.
 func (s *State) SetRemoved() {
+	s.ErrorMsg = ""
```

Comment:
Is it necessary to clear the error?

Reply:
The reason to clear it here is, in case the container that failed to start is being removed, this error might not be empty and therefore the client will get a false positive.

Reply:
Alternatively, we can add another field, like `RemovalError` (to distinguish it from `StartError` which is currently named `Error`).